From cfba2d64222a7ccfa00660ad0598cefd75669151 Mon Sep 17 00:00:00 2001
From: Andre Amegah <andrex.amegah@intel.com>
Date: Fri, 30 Nov 2012 09:26:02 +0200
Subject: [PORT FROM MAIN] Telephony: Support for stk user activity event

BZ: 61209

Stk User Activity Event is not supported in Android
framework.

This patch provides support for User Activity event.

User activity event is an STK Event Dowload commands.
When UICC registers to this event via a Setup Event List
command and next the terminal detects any user key-press
or screen-touch, the terminal informs UICC through an Stk
Envelope command. This is oneshot event as after sending
the envelope, User Activity event is removed from Event
List.

Change-Id: Ieb67f6c86a288cd25cb23a6a10bf94a4e731dd69
Orig-Change-Id: I36d15190b7d15b70850d691f307157bb807bbc9c
Category: aosp improvement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Andre Amegah <andrex.amegah@intel.com
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../internal/telephony/cat/AppInterface.java       |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/cat/AppInterface.java b/src/java/com/android/internal/telephony/cat/AppInterface.java
index 01b86ef..02481df 100644
--- a/src/java/com/android/internal/telephony/cat/AppInterface.java
+++ b/src/java/com/android/internal/telephony/cat/AppInterface.java
@@ -32,6 +32,14 @@ public interface AppInterface {
     public static final String CAT_SESSION_END_ACTION =
                                     "android.intent.action.stk.session_end";
 
+    // Broadcast to be sent to Launcher to request user activity action
+    public static final String CHECK_USER_ACTIVITY_ACTION =
+            "android.intent.action.launcher.user_activity";
+
+    // Broadcast to receive from Launcher when user activity occurs
+    public static final String USER_ACTIVITY_AVAILABLE_ACTION =
+            "android.intent.action.stk.user_activity_available";
+
     /*
      * Callback function from app to telephony to pass a result code and user's
      * input back to the ICC.
-- 
1.7.4.1

